server/tserver/src/main/java/org/apache/accumulo/tserver/compactions/CompactionExecutor.java

```diff
@@ -94,8 +112,11 @@ public boolean cancel(Status expectedStatus) {
         canceled = status.compareAndSet(expectedStatus, Status.CANCELED);
       }
 
+      if (canceled)
+        queuedTask.remove(this);
+
       if (canceled && cancelCount.incrementAndGet() % 1024 == 0) {
-        // nMeed to occasionally clean the queue, it could have canceled task with low priority that
+        // need to occasionally clean the queue, it could have canceled task with low priority that
         // hang around. Avoid cleaning it every time something is canceled as that could be
         // expensive.
```

Comment:
This sentence could be cleaned up to make it more understandable. What does "it" refer to? And does it mean it could "contain a canceled task", or that it could "have canceled a task"?

Reply:
it refers to the queue.. I can clean this up

Reply:
Changed comment in 79611ee

Reply:
Awesome, thanks! A minor cleanup would be great!